### DISCUSSION TITLE
nvim-java not working on raspberry Pi

### DISCUSSION THREAD

**Discussion by junkerjoerg12** [nvim-java/nvim-java#261] [Q&A]
I have neovim set up on my main mashine and everything works as intendet. Now I wanted to use Java on my Raspberry Pi. So I installed everything and cloned my nvim config from my github. But whenever I start nvim I get two errors showing up
First "module 'java-core.utils.notify' not found: "
![Bildschirmfoto vom 2024-07-12 14-12-14](https://github.com/user-attachments/assets/56915070-4087-4ce0-acf3-e885709d7103)
And then "Failed to source {Path/to/mynvim-java/config}
![Bildschirmfoto vom 2024-07-12 14-14-34](https://github.com/user-attachments/assets/56dbf803-9b54-4499-af88-83654b55d552)

I have already tried to uninstall the plugin and dap-core and reinstall it but that didnt change anything

I would appreciate any help!

Ps there is still some auto fill (probably not quite the correct word). For  example if i type main and then enter it does the public static void main for me , but my guess it that comes already build in with nvim.

**Comment by s1n7ax** [comment]
What plugin manager do you use?

**Reply by junkerjoerg12** [reply]
Im using lazyvim so lazy.nvim.

**Reply by s1n7ax** [reply]
Update lazy.nvim and next time you launch everything should work

**Reply by junkerjoerg12** [reply]
Well now I get two warnings: both jdtls and spring boot quit with exit code 2 and signal 0.

**Reply by s1n7ax** [reply]
Oh wait. It's ARM isn't it? Actually we don't download the compatible java version for ARM through mason. This could be the issue. To check the actual issue you can check the logs in ~/.local/state/nvim/lsp.log

**Reply by s1n7ax** [reply]
Unfortunately we don't have jdtls link either. Nether the main mason registry. I don't know if that's even possible in mason spec.

https://github.com/s1n7ax/mason-registry/blob/7162edee65a3da4e2220edcb614395c7a86f4e16/packages/jdtls/package.yaml?plain=1#L16-L27

https://github.com/s1n7ax/mason-registry/blob/7162edee65a3da4e2220edcb614395c7a86f4e16/packages/openjdk-17/package.yaml?plain=1#L18-L31'

**Reply by s1n7ax** [reply]
They do have `linux_arm` and `linux_arm64` targets. This is feasible. If you are willing to contribute, I can guide you.

https://github.com/mason-org/mason-registry/blob/e2b806916280a77e85b7ec8d8c441390509b1b90/packages/gh/package.yaml?plain=1#L31-L38

**Reply by junkerjoerg12** [reply]
I am willing to contribute but won't be available for the next week and don't have any experience in open source projects.

**Reply by s1n7ax** [reply]
I don't have time right now but I'll work on this. Would be amazing to have this running on ARM.
